convert2rhel/backup.py

```diff
@@ -367,12 +367,19 @@ def enable(self):
 
             except (OSError, IOError) as err:
                 # IOError for py2 and OSError for py3
-                loggerinst.critical("Error(%s): %s" % (err.errno, err.strerror))
+
+                loggerinst.critical_no_exit("Error(%s): %s" % (err.errno, err.strerror))
+                raise exceptions.CriticalError(
+                    id_="FAILED_TO_SAVE_FILE_TO_BACKUP_DIR",
+                    title="Failed to copy file to the backup directory.",
+                    description="During convert2rhel's tests of whether it is likely to succeed, some files on the system are changed. To enable rollback in case we detect something that we would fail to do, we backup those files prior to changing them. In the current case, we encountered a failure while performing that backup so it is unsafe to continue.",
+                    diagnosis="Failed to backup %s. Errno: %s, Error: %s" % (self.filepath, err.errno, err.strerror),
+                )
```

Comment:
@hosekadam since you worked on making the `NewRestorableFile` class, could you take a look at this and tell me if it is correct to be this way? 

I'm not sure if we should raise loggerinst.critical() here as it was before, or, if we should go with `CriticalError`.

Reply:
The way you changed it looks better and more systematic to me. So I would say it's correct.

I think @SpyTec commented this part here: https://github.com/oamg/convert2rhel/pull/875/files#r1403533541 (not it's linked to different code, but in previous I think it was related to this).